Automattic/wp-calypso
# Customize site list styling

#### Changes proposed in this Pull Request
* Turn `PluginSiteList` into a functional React component
* Update the CSS of the Site list section to match the Figma layout at oU6nEIeiKSg8CayhEtgR6b-fi-6559%3A49151

PS: `PluginSiteNetwork` handle sites with *secondary* sites but as it is n independent component we can handle it on a separate PR.

#### Testing instructions
- Go to `/plugins` without a selected site
- Click in a plugin, it will redirect you to the plugin details page
- Check if the site list layout matches the below and try to use the buttons **install**, **active**, **autoupdates** and **remove**

![Screen Shot 2021-11-17 at 18 21 09](https://user-images.githubusercontent.com/5039531/142291894-9f7d9312-f0c1-4d48-a92a-65ebd24e39f7.png)

- Select a site
- Check if the plugin is installed, if not, install
- Check if the layout matches the below

![Screen Shot 2021-11-18 at 14 46 11](https://user-images.githubusercontent.com/5039531/142485780-6020f07f-89e5-4c98-8c99-4885398e5845.png)

- Check the multi-site version on mobile or emulate it via your browsers
- Check if the layout matches the below


![Screen Shot 2021-11-23 at 13 18 12](https://user-images.githubusercontent.com/5039531/143072652-8e302067-82ae-4c16-8e88-9b004fe39b9e.png)


---

Fixes #57747
